sentenai/api/__init__.py

```diff
@@ -0,0 +1,513 @@
+        elif resp.status_code == 404:
+            raise NotFound(
+                'The stream at "/streams/{}" '
+                'does not exist'.format(stream.name, eid))
+        else:
+            status_codes(resp)
+
+        if eid is not None:
+            return {
+                'id': resp.headers['location'],
+                'ts': resp.headers['timestamp'],
+                'event': resp.json()}
+        else:
+            return resp.json()
+
+
+    def put(self, stream, event, id=None, timestamp=None):
+        """Put a new event into a stream.
+
+        Arguments:
+           stream    -- A stream object corresponding to a stream stored
+                        in Sentenai.
+           event     -- A JSON-serializable dictionary containing an
+                        event's data
+           id        -- A user-specified id for the event that is unique to
+                        this stream (optional)
+           timestamp -- A user-specified datetime object representing the
+                        time of the event. (optional)
+        """
+        headers = {
+            'content-type': 'application/json'
+        }
+        jd = event
+
+        if timestamp:
+            headers['timestamp'] = iso8601(timestamp)
+
+        if id:
+            url = '{host}/streams/{sid}/events/{eid}'.format(
+                sid=stream.name, host=self.host, eid=id
+            )
+            resp = self.session.put(url, json=jd, headers=headers)
+            if resp.status_code not in [200, 201]:
+                status_codes(resp)
+            else:
+                return id
+        else:
+            url = '{host}/streams/{sid}/events'.format(
+                sid=stream._name, host=self.host
+            )
+            resp = self.session.post(url, json=jd, headers=headers)
+            if resp.status_code in [200, 201]:
+                return resp.headers['location']
+            else:
+                status_codes(resp)
+                raise APIError(resp)
+
+
+    def streams(self, name=None, meta={}):
+        """Get list of available streams.
+
+        Optionally, parameters may be supplied to enable searching
+        for stream subsets.
+
+        Arguments:
+           name -- A regular expression pattern to search names for
+           meta -- A dictionary of key/value pairs to match from stream
+                   metadata
+        """
+        url = "/".join([self.host, "streams"])
+        resp = self.session.get(url, params={'stats': 'true'})
+        status_codes(resp)
+
+        def filtered(s):
+            f = True
+            if name:
+                f = bool(re.search(name, s['name']))
+            for k, v in meta.items():
+                f = f and s.get('meta', {}).get(k) == v
+            return f
+
+        try:
+            return StreamsView(self, resp.json())
+        except:
+            raise
+            raise SentenaiException("Something went wrong")
```

Comment:
what's the ideal behavior? it seems like `StreamsView` has no defined exceptions behavior, so should we remove this try/except handler?